charts/hdfs-namenode-k8s/templates/namenode-statefulset.yaml

```diff
@@ -94,8 +160,32 @@ spec:
               mountPath: /etc/security/
               readOnly: true
             {{- end }}
-      {{- if .Values.kerberosEnabled }}
       initContainers:
+        # TODO: Support upgrade of metadata in case a new Hadoop version requires it.
+        - name: write-start-script
+          image: busybox:1.27.1
+          command: ['sh', '-c']
+          args:
+            - >-
+              (echo '#!/bin/bash -x' > $OUT);
```

Comment:
I wonder if this helm chart can put this script in a config map instead of this init-container. That way, we can at least remove the echo part and make the script more readable. What do you think?

Reply:
I also wonder if we can have multiple scripts in the config map and switch between them as requested by the user. Say one for regular mode, another for metadata version upgrade, etc. Maybe we can even support a completely ad-hoc script using a helm variable.

Reply:
Seems this approach is not as crazy as I thought originally. See http://blog.phymata.com/2017/07/29/inject-an-executable-script-into-a-container-in-kubernetes/